Detectors/MUON/MID/Raw/include/MIDRaw/RawFileReader.h

```diff
@@ -0,0 +1,66 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file   MIDRaw/RawFileReader.h
+/// \brief  MID raw file reader
+/// \author Diego Stocco <Diego.Stocco at cern.ch>
+/// \date   25 November 2019
+#ifndef O2_MID_RAWFILEREADER_H
+#define O2_MID_RAWFILEREADER_H
+
+#include <fstream>
+#include <vector>
+#include <array>
+#include "MIDRaw/CrateParameters.h"
+#include "MIDRaw/RawBuffer.h"
+#include "MIDRaw/RawUnit.h"
+
+namespace o2
+{
+namespace mid
+{
+template <typename T>
+class RawFileReader
+{
+ public:
+  bool init(const char* inFilename, bool readContinuous = false);
```

Comment:
why an init function and not simply a constructor ?

Reply:
Because I want to be able to delay initialisation. This allows me to:
- use the same class in a loop (just change the file)
- avoid to have to construct the object in the "init" of the DPL device, since in any case all object are initialised there.